json_object_private.h

```diff
@@ -43,36 +45,53 @@ struct json_object
 	json_object_private_delete_fn *_delete;
 	json_object_to_json_string_fn *_to_json_string;
 	struct printbuf *_pb;
-	union data
-	{
-		json_bool c_boolean;
-		double c_double;
-		struct
-		{
-			union
-			{
-				int64_t c_int64;
-				uint64_t c_uint64;
-			} cint;
-			enum json_object_int_type cint_type;
-		} c_int;
-		struct lh_table *c_object;
-		struct array_list *c_array;
-		struct
-		{
-			union
-			{
-				/* optimize: if we have small strings, we can store them
-				 * directly. This saves considerable CPU cycles AND memory.
-				 */
-				char *ptr;
-				char data[LEN_DIRECT_STRING_DATA];
-			} str;
-			int len;
-		} c_string;
-	} o;
 	json_object_delete_fn *_user_delete;
 	void *_userdata;
+	char data[1]; // Actually the rest of a struct json_object_${o_type}
```

Comment:
Is there any further plan here. It looks like a redundant data.

Reply:
You're right, I'll drop this field.

Reply:
See commit 4c107121